nano/node/active_transactions.cpp

```diff
@@ -67,99 +67,152 @@ void nano::active_transactions::stop ()
 
 void nano::active_transactions::block_cemented_callback (std::shared_ptr<nano::block> const & block_a)
 {
-	auto transaction = node.store.tx_begin_read ();
+	nano::store::read_transaction transaction = node.store.tx_begin_read ();
+	auto status_type = determine_election_status (transaction, block_a);
 
-	boost::optional<nano::election_status_type> election_status_type;
-	if (!confirmation_height_processor.is_processing_added_block (block_a->hash ()))
+	if (!status_type)
+		return;
+
+	if (*status_type == nano::election_status_type::inactive_confirmation_height)
```

Comment:
I think a switch statement would be better here, generally better for any enum value. Then the compiler can warn if there are missing cases which is important when we add new values to enums.